contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6180,6 +6183,36 @@ transformSelectIntoStmt(CreateTableAsStmt *stmt)
 		foreach (elements, q->targetList)
 		{
 			TargetEntry *tle = (TargetEntry *)lfirst(elements);
+			char *original_name = NULL;
+
+			if (tle->resname != NULL && !tle->resjunk && strlen(tle->resname) < NAMEDATALEN - 1)
+			{	
+
+				original_name = tle->resname;
+				tle->resname = downcase_identifier(tle->resname, strlen(tle->resname), false, false);
```

Comment:
resname should be downcased irrespective of whether we can find original_name or not or in other words, down-casing should not get affect by whether or not `strlen(tle->resname) < NAMEDATALEN - 1`